cylc/flow/tests/test_pathutil.py

```diff
@@ -0,0 +1,145 @@
+        # tail1 = expected tail of return value from configuration
+        # args = extra *args
+        # tail2 = expected tail of return value from extra args
+        for func, cfg, tail1 in (
+            (get_suite_run_dir, 'run directory', ''),
+            (get_suite_run_job_dir, 'run directory', '/log/job'),
+            (get_suite_run_log_dir, 'run directory', '/log/suite'),
+            (get_suite_run_rc_dir, 'run directory', '/log/suiterc'),
+            (get_suite_run_share_dir, 'work directory', '/share'),
+            (get_suite_run_work_dir, 'work directory', '/work'),
+        ):
+            for args, tail2 in (
+                ((), ''),
+                (('comes', 'true'), '/comes/true'),
+            ):
+                self.assertEqual(
+                    f'/home/sweet/cylc-run/my-suite/dream{tail1}{tail2}',
+                    func('my-suite/dream', *args),
+                )
+                mocked.get_host_item.assert_called_with(cfg)
+                mocked.get_host_item.reset_mock()
+
+    @patch('cylc.flow.pathutil.glbl_cfg')
+    def test_get_suite_run_names(self, mocked_glbl_cfg):
+        """Usage of get_suite_run_*name."""
+        mocked = MagicMock()
+        mocked_glbl_cfg.return_value = mocked
+        mocked.get_host_item.return_value = '/home/sweet/cylc-run'
+        # func = get_remote_* function to test
+        # cfg = configuration used in mocked global configuration
+        # tail1 = expected tail of return value from configuration
+        for func, cfg, tail1 in (
+            (get_suite_run_log_name, 'run directory', '/log/suite/log'),
+            (get_suite_run_pub_db_name, 'run directory', '/log/db'),
+        ):
+            self.assertEqual(
+                f'/home/sweet/cylc-run/my-suite/dream{tail1}',
+                func('my-suite/dream'),
+            )
+            mocked.get_host_item.assert_called_with(cfg)
+            mocked.get_host_item.reset_mock()
+
+    @patch('cylc.flow.pathutil.os.makedirs')
+    @patch('cylc.flow.pathutil.glbl_cfg')
+    def test_make_suite_run_tree(self, mocked_glbl_cfg, mocked_makedirs):
+        """Usage of make_suite_run_tree."""
+        mocked = MagicMock()
+        mocked_glbl_cfg.return_value = mocked
+        mocked.get_host_item.return_value = '/home/sweet/cylc-run'
+        mocked_cfg = MagicMock()
+        mocked_cfg['run directory rolling archive length'] = 0
+        mocked.get.return_value = mocked_cfg
+        make_suite_run_tree('my-suite/dream')
+        self.assertEqual(mocked_makedirs.call_count, 6)
+        mocked_makedirs.assert_has_calls((
+            call(f'/home/sweet/cylc-run/my-suite/dream{tail}', exist_ok=True)
+            for tail in (
+                '',
+                '/log/suite',
+                '/log/job',
+                '/log/suiterc',
+                '/share',
+                '/work',
+            )
+        ))
+
+
+if __name__ == "__main__":
+    unittest.main()
```

Comment:
If this is executed, this will result in an error, as `unittest` was not imported`. Maybe add `import main` to the existing `from unittest` import?

Reply:
@kinow Thanks for spotting this. I have corrected this just now.